### PR TITLE
Translation for fr/extractor-objects

### PR DESCRIPTION
**Pull request by apointeau**

### CODE REVIEW

File: _fr/tour/extractor-objects.md [hunk 1]
```diff
@@ -3,10 +3,64 @@ layout: tour
...
+num: 18
 
 language: fr
 
-next-page: generic-classes
+next-page: for-comprehensions
 previous-page: regular-expression-patterns
 ---
+
+Un objet extracteur est un objet avec une méthode `unapply`. Tandis que la méthode `apply` ressemble à un constructeur qui prend des arguments et crée un objet, `unapply` prend un object et essaye de retourner ses arguments. Il est utilisé le plus souvent en filtrage par motif ou avec les fonctions partielles.
```
Comment (sjrd) [inline _fr/tour/extractor-objects.md:14]:
```suggestion
Un objet extracteur est un objet avec une méthode `unapply`. Tandis que la méthode `apply` ressemble à un constructeur qui prend des arguments et crée un objet, `unapply` prend un object et essaye de retourner ses arguments. Il est utilisé le plus souvent en filtrage par motif (*pattern matching*) ou avec les fonctions partielles.
```
I did not know the French term "filtrage par motif". I always use "pattern matching" even when I speak in French. I wouldn't be surprised if I'm not alone. So I suggest to at least once write the English name.

File: _fr/tour/extractor-objects.md [hunk 2]
```diff
@@ -3,10 +3,64 @@ layout: tour
...
+
+S'il n'y a pas de correspondance, une `scala.MatchError` est levée :
+
+```scala
+val CustomerID(name3) = "-asdfasdfasdf"
+```
+
+Le type de retour de `unapply` doit être choisi comme suit :
+
+* Si c'est juste un test, retourner un `Boolean`. Pour l'instance `case even()`.
```
Comment (sjrd) [inline _fr/tour/extractor-objects.md:60]:
```suggestion
* Si c'est juste un test, retourner un `Boolean`. Par exemple, `case even()`.
```

File: _fr/tour/extractor-objects.md [hunk 3]
```diff
@@ -3,10 +3,64 @@ layout: tour
...
+* Si cela retourne une seule sous-valeur de type T, retourner un `Option[T]`.
+* Si vous souhaitez retourner plusieurs sous-valeurs `T1,...,Tn`, groupez-les dans un tuple optionnel `Option[(T1,...,Tn)]`.
+
+Parfois, le nombre de valeurs à extraire n'est pas fixe et on souhaiterait retourner un nombre arbitraire de valeurs, en fonction des données d'entrée. Pour ce cas, vous pouvez définir des extracteurs avec la méthode `unapplySeq` qui retourne un `Option[Seq[T]]`. Un exemple commun d'utilisation est la déconstruction d'une liste en utilisant `case List(x, y, z) =>` et en décomposant une `String` en utilisant une expression régulière `Regex`, comme `case r(name, remainingFields @ _*) =>`.
```
Comment (sjrd) [inline _fr/tour/extractor-objects.md:64]:
```suggestion
Parfois, le nombre de valeurs à extraire n'est pas fixe et on souhaiterait retourner un nombre arbitraire de valeurs, en fonction des données d'entrée. Pour ce cas, vous pouvez définir des extracteurs avec la méthode `unapplySeq` qui retourne un `Option[Seq[T]]`. Un exemple commun d'utilisation est la déconstruction d'une liste en utilisant `case List(x, y, z) =>`. Un autre est la décomposition d'une `String` en utilisant une expression régulière `Regex`, comme `case r(name, remainingFields @ _*) =>`.
```

**Review by sjrd — CHANGES_REQUESTED** [review]
Thanks for the PR. Sorry it took so long for someone to have a look at it.

**Comment by SethTisue** [comment]
@apointeau care to respond to the review feedback?

**Comment by apointeau** [comment]
> @apointeau care to respond to the review feedback?

Good to me, thank you @sjrd

**Review by sjrd — APPROVED** [review]

**Comment by SethTisue** [comment]
No policy. I often squash, but sometimes regular-merge if the commits seem history-conscious.

**Pull request merged**